modules/core/04-channel/v2/keeper/keeper.go

```diff
@@ -61,3 +62,103 @@ func (k *Keeper) GetCounterparty(ctx context.Context, clientID string) (types.Co
 	k.cdc.MustUnmarshal(bz, &counterparty)
 	return counterparty, true
 }
+
+// GetPacketReceipt returns the packet receipt from the packet receipt path based on the sourceID and sequence.
+func (k *Keeper) GetPacketReceipt(ctx context.Context, sourceID string, sequence uint64) (string, bool) {
+	store := k.storeService.OpenKVStore(ctx)
+	bigEndianBz := sdk.Uint64ToBigEndian(sequence)
+	bz, err := store.Get(hostv2.PacketReceiptKey(sourceID, bigEndianBz))
+	if err != nil {
+		panic(err)
+	}
+	if len(bz) == 0 {
+		return "", false
+	}
+	return string(bz), true
+}
+
+// SetPacketReceipt writes the packet receipt under the receipt path
+// This is a public path that is standardized by the IBC V2 specification.
+func (k *Keeper) SetPacketReceipt(ctx context.Context, sourceID string, sequence uint64) {
+	store := k.storeService.OpenKVStore(ctx)
+	bigEndianBz := sdk.Uint64ToBigEndian(sequence)
+	if err := store.Set(hostv2.PacketReceiptKey(sourceID, bigEndianBz), []byte{byte(1)}); err != nil {
+		panic(err)
+	}
+}
+
+// SetPacketAcknowledgement writes the acknowledgement hash under the acknowledgement path
+// This is a public path that is standardized by the IBC V2 specification.
+func (k *Keeper) SetPacketAcknowledgement(ctx context.Context, sourceID string, sequence uint64, ackHash []byte) {
+	store := k.storeService.OpenKVStore(ctx)
+	bigEndianBz := sdk.Uint64ToBigEndian(sequence)
+	if err := store.Set(hostv2.PacketAcknowledgementKey(sourceID, bigEndianBz), ackHash); err != nil {
+		panic(err)
+	}
+}
+
+// HasPacketAcknowledgement check if the packet ack hash is already on the store.
+func (k *Keeper) HasPacketAcknowledgement(ctx context.Context, sourceID string, sequence uint64) bool {
+	store := k.storeService.OpenKVStore(ctx)
+	bigEndianBz := sdk.Uint64ToBigEndian(sequence)
+	found, err := store.Has(hostv2.PacketAcknowledgementKey(sourceID, bigEndianBz))
+	if err != nil {
+		panic(err)
+	}
+
+	return found
+}
+
+// GetPacketCommitment returns the packet commitment hash under the commitment path.
+func (k *Keeper) GetPacketCommitment(ctx context.Context, sourceID string, sequence uint64) (string, bool) {
+	store := k.storeService.OpenKVStore(ctx)
+	bigEndianBz := sdk.Uint64ToBigEndian(sequence)
```

Comment:
don't want to be nitting too hard but might be nicer to do this inside `host` pkg. API looks friendlier with a uint64 sequence, and the formatting is a `host` responsibility imo 🤷🏻